### PR TITLE
[bugfix] Fixing #651 to remove compiler warnings

### PR DESCRIPTION
**Pull request by brianjjones**
Also fixing the return type to be int, since that's what
zephyr returns.

### CODE REVIEW

File: src/zjs_i2c.c [hunk 1]
```diff
@@ -169,13 +175,13 @@ static jerry_value_t zjs_i2c_write(const jerry_value_t function_obj,
...
-    uint8_t error_msg = zjs_i2c_handle_write((uint8_t)bus,
-                                             dataBuf->buffer,
-                                             dataBuf->bufsize,
-                                             (uint16_t)address);
+    int error_msg = zjs_i2c_handle_write((uint8_t)bus,
+                                         dataBuf->buffer,
+                                         dataBuf->bufsize,
+                                         (uint16_t)address);
 
     return jerry_create_number(error_msg);
```
Comment (grgustaf) [inline src/zjs_i2c.c]: This should really return an error object back to JavaScript instead I think? The API documentation seems to say this function is void so it should return undefined in the success case.

File: src/zjs_i2c.c [hunk 2]
```diff
@@ -73,17 +73,23 @@ static jerry_value_t zjs_i2c_read_base(const jerry_value_t this,
...
-        uint8_t error_msg = zjs_i2c_handle_read((uint8_t)bus,
-                                                buf->buffer,
-                                                buf->bufsize,
-                                                (uint16_t)address);
+        int error_msg = zjs_i2c_handle_read((uint8_t)bus,
+                                            buf->buffer,
+                                            buf->bufsize,
+                                            (uint16_t)address);
+        if (error_msg != 0) {
+            ERR_PRINT("i2c_read failed with error %i\n", error_msg);
```
Comment (brianjjones) [inline src/zjs_i2c.c]: I'd like to give the user the ability to recover and try again if it should i2c_read fail.  It happens (rarely) but I have seen it happen over long periods of time.  That's why I have an ERR_PRINT vs zjs_error

**Comment by grgustaf** [comment]
+1, merging as is.

File: src/zjs_i2c.c (hunk 2 above)
Comment (grgustaf) [inline src/zjs_i2c.c]: They can still catch the error with try/catch.

**Pull request merged**